plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/ExamplePluginConfiguration.java

```diff
@@ -19,51 +19,35 @@
 
 package org.elasticsearch.plugin.example;
 
-import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.common.xcontent.yaml.YamlXContent;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 
 import java.io.IOException;
 import java.nio.file.Path;
 
-import static java.nio.charset.StandardCharsets.UTF_8;
-import static java.nio.file.Files.newBufferedReader;
-import static org.elasticsearch.common.io.Streams.copyToString;
-
 /**
  * Example configuration.
  */
 public class ExamplePluginConfiguration {
-    private String test = "not set in config";
-
+    
+    private static final String TEST_KEY = "test";
+    
+    private static final String NO_SETTING_FOUND = "not set in config";
+    
+    private static final String CONFIG_FILE = "jvm-example/example.yaml";
+    
+    private Settings settings;
+    
     @Inject
     public ExamplePluginConfiguration(Environment env) throws IOException {
+    
         // The directory part of the location matches the artifactId of this plugin
-        Path configFile = env.configFile().resolve("jvm-example/example.yaml");
-        String contents = copyToString(newBufferedReader(configFile, UTF_8));
-        XContentParser parser = YamlXContent.yamlXContent.createParser(contents);
-
-        String currentFieldName = null;
-        XContentParser.Token token = parser.nextToken();
-        assert token == XContentParser.Token.START_OBJECT;
-        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
-            if (token == XContentParser.Token.FIELD_NAME) {
-                currentFieldName = parser.currentName();
-            } else if (token.isValue()) {
-                if ("test".equals(currentFieldName)) {
-                    test = parser.text();
-                } else {
-                    throw new ElasticsearchParseException("Unrecognized config key: {}", currentFieldName);
-                }
-            } else {
-                throw new ElasticsearchParseException("Unrecognized config key: {}", currentFieldName);
-            }
-        }
+        Path path = env.configFile().resolve(CONFIG_FILE);
+        settings = Settings.builder().loadFromPath(path).build();
     }
 
     public String getTestConfig() {
-        return test;
+        return settings.get(TEST_KEY, NO_SETTING_FOUND);
```

Comment:
I mean maybe instead of declaring the key and the default we should instead declare a `org.elasticsearch.common.settings.Setting` and use it. Like, for example, AutoCreateIndex#AUTO_CREATE_INDEX_SETTING. That might mean moving the setting out of this configuration class and into some other place to feel more natural, but that is the whole point of the jvm-example plugin - to have a working semi natural plugin.